File: repos_estudiantes_ALVAREZ_primera_entrega/CORY_CORNEJO_ALEJANDRO/Herencia/Ejer3/P3E3/src/main/java/com/mycompany/p3e3/P3E3.java
package com.mycompany.p3e3;
public class P3E3 {

    public static void main(String[] args) {
        Animal[] animales = new Animal[5];
        animales[0] = new Leon("Simba", 5);
        animales[1] = new Pinguino("Pingu", 3);
        animales[2] = new Canguro("Kangy", 4);
        animales[3] = new Leon("Nala", 6);
        animales[4] = new Pinguino("Glacier", 2);

        System.out.println("Desplazamientos:");
        for (Animal a : animales) {
            a.desplazarse();
        }
    }
}

File: repos_estudiantes_ALVAREZ_primera_entrega/CORY_CORNEJO_ALEJANDRO/Herencia/Ejer3/P3E3.py
class Animal:
    def __init__(self, nombre, edad):
        self.nombre = nombre
        self.edad = edad

    def desplazarse(self):
        print(f"{self.nombre} se desplaza de forma genérica.")
class Leon(Animal):
    def desplazarse(self):
        print(f"{self.nombre} camina con fuerza por la sabana.")

class Pinguino(Animal):
    def desplazarse(self):
        print(f"{self.nombre} nada ágilmente en aguas frías.")

class Canguro(Animal):
    def desplazarse(self):
        print(f"{self.nombre} salta velozmente por la pradera.")
class Main():
    animales = [
        Leon("Simba", 5),
        Pinguino("Pingu", 3),
        Canguro("Kangy", 4),
        Leon("Nala", 6),
        Pinguino("Glacier", 2)
    ]

    print("Desplazamientos:")
    for animal in animales:
        animal.desplazarse()
